cumulusci/tasks/metadata_etl/tests/test_field_sets.py

```diff
@@ -0,0 +1,157 @@
+        </displayedFields>
+        <displayedFields>
+            <field>ClosedDate</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>Branch__c</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <displayedFields>
+            <field>Severity__c</field>
+            <isFieldManaged>false</isFieldManaged>
+            <isRequired>false</isRequired>
+        </displayedFields>
+        <label>Incident Detail</label>
+    </fieldSets>
+</CustomObject>
+"""
+
+
+class TestAddFieldsToFieldSet:
+    def test_adds_fields(self):
+        task = create_task(
+            AddFieldsToFieldSet,
+            {
+                "field_set": "Case.IncidentDetail",
+                "fields": ["IncidentDateTime__c", "LocationNotes__c"],
+            },
+        )
+
+        # Validate that the fields have been added as displayedFields
+        tree = metadata_tree.fromstring(OBJECT_XML)
+        result = task._transform_entity(tree, "Case")
+        print(result.tostring())
+        fields = result.find("fieldSets", fullName="IncidentDetail").findall(
+            "displayedFields"
+        )
+        print(len(fields))
+
+        test_elem = next(f for f in fields if f.field.text == "IncidentDateTime__c")
+        assert test_elem is not None
+        assert test_elem.isFieldManaged.text == "false"
+        assert test_elem.isRequired.text == "false"
+
+        test_elem = next(f for f in fields if f.field.text == "LocationNotes__c")
+        assert test_elem is not None
+        assert test_elem.isFieldManaged.text == "false"
+        assert test_elem.isRequired.text == "false"
+
+        # Validate that the fields were removed from availableFields
+        fields = result.find("fieldSets", fullName="IncidentDetail").findall(
+            "availableFields"
+        )
+        assert "IncidentDateTime__c" not in (f.field.text for f in fields)
+        assert "LocationNotes__c" not in (f.field.text for f in fields)
+
+        # Validate that the other field remains in availableFields
+        assert "WatchList__c" in (f.field.text for f in fields)
+
+    def test_does_not_add_duplicate_values(self, caplog):
+        task = create_task(
+            AddFieldsToFieldSet,
+            {
+                "field_set": "Case.IncidentDetail",
+                "fields": ["ContactId"],
+            },
+        )
+
+        caplog.set_level(logging.WARNING)
+
+        # Validate that the duplicate entry is not added
+        tree = metadata_tree.fromstring(OBJECT_XML)
+        result = task._transform_entity(tree, "Case")
+        fields = result.find("fieldSets", fullName="IncidentDetail").findall(
+            "displayedFields"
+        )
+
+        test_elem = list(f for f in fields if f.field.text == "ContactId")
```

Comment:
You can just use a list comprehension here rather than converting a generator expression to a list.